kpi/views/v2/asset.py

```diff
@@ -618,6 +620,68 @@ def metadata(self, request):
         metadata = self.get_metadata(queryset)
         return Response(metadata)
 
+    @action(detail=False, methods=['GET'],
+            renderer_classes=[renderers.JSONRenderer],
+            url_path=r'migrate(?:/(?P<task_id>[\d\w\-]+))?')
+    def migrate(self, request, task_id: str = None, **kwargs):
+        """
+
+        Really basic endpoint what allows super users to migration projects
+        from KoBoCAT.
+
+        1. Call this endpoint with `?username=<username>`
+        2. Fetch url provided to check the state of the Celery task.
+           It can be:
+            - 'PENDING'
+            - 'FAILED'
```

Comment:
I forgot to restart my Celery workers when testing locally, so my attempt failed with `TypeError: sync_kobocat_xforms() got an unexpected keyword argument 'populate_xform_kpi_asset_uid'`. However, the status never changed from `PENDING` to `FAILED` (or anything else).

I know this is a throw-away, temporary endpoint, but I guess we should be aware that the Celery `res.state` isn't too reliable—unless it's waiting several minutes to retry the task(?)

Reply:
I updated the comments